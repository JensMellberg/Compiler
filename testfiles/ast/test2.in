int a(int x){
y = 5;
if(1){
	x = 5;
	x = y;
	}
}
