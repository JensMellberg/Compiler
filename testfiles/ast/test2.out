Program
  List
    FuncDecl
      IdDecl ID="a"
      List
        IdParam
          IdDecl ID="x"
      Block
        List
          Assign
            IdUse ID="y"
            Numeral NUMERAL="5"
          If
            Numeral NUMERAL="1"
            Block
              List
                Assign
                  IdUse ID="x"
                  Numeral NUMERAL="5"
                Assign
                  IdUse ID="x"
                  IdUse ID="y"
            Opt
