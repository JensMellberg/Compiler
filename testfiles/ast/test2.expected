Program
  List
    FuncDecl
      IdDecl ID="a"
      List
      Block
        List
          IdReadDecl
            IdDecl ID="x"
            Opt
          IdReadDecl
            IdDecl ID="y"
            Opt
          Assign
            IdUse ID="x"
            Numeral NUMERAL="5"
          Assign
            IdUse ID="x"
            IdUse ID="y"
